src/sentry/integrations/source_code_management/webhook.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from collections.abc import Mapping
+from typing import Any
+
+from sentry.integrations.utils.metrics import IntegrationWebhookEventType
+from sentry.models.repository import Repository
+
+
+class SCMWebhook(ABC):
+    @property
+    @abstractmethod
+    def provider(self) -> str:
+        raise NotImplementedError
```

Comment:
I wonder if this is something we can commonly extract from the `IntegrationInstallation.provider` instead of having each one provide its own string.

Reply:
there is only one webhook that processes stuff for all installations, but i can see